crates/next-api/src/app.rs

```diff
@@ -1245,22 +1252,52 @@ impl AppEndpoint {
 
                 let EntryChunkGroupResult {
                     asset: rsc_chunk, ..
-                } = *{
-                    let _span = tracing::trace_span!("server node entrypoint").entered();
-                    chunking_context.entry_chunk_group(
-                        server_path.join(
-                            format!(
-                                "app{original_name}.js",
-                                original_name = app_entry.original_name
-                            )
-                            .into(),
-                        ),
-                        app_entry.rsc_entry,
-                        Vc::cell(evaluatable_assets),
-                        Value::new(AvailabilityInfo::Root),
-                    )
+                } = *(async {
+                    let mut current_chunks = OutputAssets::empty();
+                    let mut current_availability_info = AvailabilityInfo::Root;
+                    if let Some(client_references) = client_references {
+                        for server_component in
+                            client_references.server_component_entries.iter().copied()
+                        {
+                            let server_path = server_component.server_path();
+                            let is_layout =
+                                server_path.file_stem().await?.as_deref() == Some("layout");
+
+                            let chunk_group = chunking_context
+                                .chunk_group(
+                                    server_component.ident(),
+                                    Vc::upcast(server_component),
+                                    Value::new(current_availability_info),
+                                )
+                                .await?;
+
+                            if is_layout {
+                                current_chunks = current_chunks
+                                    .concatenate(chunk_group.assets)
+                                    .resolve()
+                                    .await?;
```

Comment:
That `resolve` is also done here, but I don't entirely know why/if it's better this way
https://github.com/vercel/next.js/blob/aceeaea4cde95ce4a44b2dfaaa9bc73fe3f527b5/crates/next-core/src/next_app/app_client_references_chunks.rs#L252-L254